kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -62,7 +63,7 @@ trait ProcBuilder {
   def mainResource: Option[String] = {
     // 1. get the main resource jar for user specified config first
     // TODO use SPARK_SCALA_VERSION instead of SCALA_COMPILE_VERSION
-    val jarName = s"${module}_$SCALA_COMPILE_VERSION-$KYUUBI_VERSION.jar"
+    val jarName = s"${module}_$scalaRuntimeSemanticVersion-$KYUUBI_VERSION.jar"
```

Comment:
The engine's Scala runtime version is independent of the server, e.g. it allows building a server with Scala 2.12, but using Spark with Scala 2.13.

Reply:
No problem, this change is intended to pass the tests on the same scala version first.
e.g. Server on Scala 2.13+ Spark on Scala 2.13

Reply:
For the dynamic combination, we could do that in follow-ups.

Reply:
The quick idea in my mind is extracting the `SPARK_SCALA_VERSION` from `$SPARK_HOME/jars/spark-core_{scala_binary_version}-{spark_version}.jar`

Reply:
Adopted.